### PR TITLE
Add sort icon and update guidance on icons

### PR DESCRIPTION
**Pull request by MisRob**
## Description

Adds sort icon that is needed in Studio.

I also documented the steps for adding a new icon and rewrote icons documentation. I didn't update parts that will soon be obsolete (3rd party icons, https://github.com/learningequality/kolibri-design-system/issues/540) and removed them instead. 

#### Issue addressed

- Recently, one of the contributors tried to add a new icon and the process wasn't clear from the current documentation
- KDS part of Studio's https://github.com/learningequality/studio/issues/4426
- Updates large portion of documentation for https://github.com/learningequality/kolibri-design-system/issues/540

### Before/after screenshots

Screenshot from 2024-02-07 11-00-47

## Changelog

- [#542]
  - **Description:** Adds a new `sort` icon
  - **Products impact:** New icon
  - **Addresses:** https://github.com/learningequality/studio/issues/4426
  - **Components:** -
  - **Breaking:** no
  - **Impacts a11y:** -
  - **Guidance:** -

[#542]: (https://github.com/learningequality/kolibri-design-system/pull/542

- [#542]
  - **Description:** Updates documentation for icons to the new process, adds clear guidelines
  - **Products impact:** None
  - **Addresses:** -
  - **Components:** -
  - **Breaking:** -
  - **Impacts a11y:** -
  - **Guidance:** -

[#542]: (https://github.com/learningequality/kolibri-design-system/pull/542

## Steps to test

- Preview the icon: https://deploy-preview-542--kolibri-design-system.netlify.app/icons/#token-sort
- Is the documentation page clear and easy to follow? https://github.com/learningequality/kolibri-design-system/blob/80ee8eda205b38d0aa84a640a7bf75eaf8a505bf/dev_docs/05_icons.md

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] ~Critical and brittle code paths are covered by unit tests~
- [x] The change is described in the changelog section above

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] ~Are there tests for this change?~
- [ ] ~Are all UI components LTR and RTL compliant (if applicable)?~

## After review

- [x] The changelog item has been pasted to the `CHANGELOG.md`

### CODE REVIEW

File: dev_docs/05_icons.md [hunk 1]
```diff
@@ -1,48 +1,31 @@
...
 
-```bash
-yarn run precompile-custom-svgs
-```
+## How to add a new icon
 
-One of these commands should be run after any icon changes.
+1. Paste a svg file to [`custom-icons`](../custom-icons/)
+2. Run `yarn run precompile-custom-svgs` to generate a corresponding Vue component. The generated component will be saved to [`lib/KIcon/precompiled-icons/le`](../lib/KIcon/precompiled-icons/le)
+3. Add a new entry for the generated component to [the icons definitions file](../lib/KIcon/iconDefinitions.js). Choose a unique alias that describes the purpose of the icon in our products well.
```
Comment (AlexVelezLl) [inline dev_docs/05_icons.md]: I think would be great to explain here the object template with some example, like that the `icon` property is the imported vue file, and other properties as `rtlFlip`, `defaultColor`, etc.

**Review by AlexVelezLl — COMMENTED** [review]
It is looking great  =) Docs are much more clearer now :tada:

File: dev_docs/05_icons.md (hunk 1 above)
Comment (MisRob) [reply dev_docs/05_icons.md]: Yes, that's a good idea, thank you. I will add a bit of guidance on these too.

**Comment by MisRob** [comment]
Thank you, @AlexVelezLl. Updated.

**Review by nucleogenesis — APPROVED** [review]
Changes LGTM for both the icon & the docs.

The doc updates set us up perfectly for the work we'll be doing in #540 😃

**Pull request merged**